slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala

```diff
@@ -59,63 +60,71 @@ private[slf4j] object Slf4jLoggerInternal {
     )
   }
 
-  final class Slf4jLogger[F[_]](val logger: JLogger)(implicit F: Sync[F])
-      extends SelfAwareStructuredLogger[F] {
-    override def isTraceEnabled: F[Boolean] = F.delay(logger.isTraceEnabled)
-    override def isDebugEnabled: F[Boolean] = F.delay(logger.isDebugEnabled)
-    override def isInfoEnabled: F[Boolean] = F.delay(logger.isInfoEnabled)
-    override def isWarnEnabled: F[Boolean] = F.delay(logger.isWarnEnabled)
-    override def isErrorEnabled: F[Boolean] = F.delay(logger.isErrorEnabled)
+  final class Slf4jLogger[F[_]](val logger: JLogger, sync: Sync.Type = Sync.Type.Delay)(implicit
+      F: Sync[F]
+  ) extends SelfAwareStructuredLogger[F] {
+
+    @deprecated("Use constructor with sync", "2.6.0")
+    def this(logger: JLogger)(
+        F: Sync[F]
+    ) =
+      this(logger, Sync.Type.Delay)(F)
+
+    override def isTraceEnabled: F[Boolean] = F.suspend(sync)(logger.isTraceEnabled)
+    override def isDebugEnabled: F[Boolean] = F.suspend(sync)(logger.isDebugEnabled)
+    override def isInfoEnabled: F[Boolean] = F.suspend(sync)(logger.isInfoEnabled)
+    override def isWarnEnabled: F[Boolean] = F.suspend(sync)(logger.isWarnEnabled)
+    override def isErrorEnabled: F[Boolean] = F.suspend(sync)(logger.isErrorEnabled)
```

Comment:
I wonder if these ones should be kept as `F.delay(...)`.

1. I understand that actually logging may be blocking, but I don't think I heard of a logger where checking the configuration is blocking.

2. I guess a client may check these to avoid doing some expensive work if logging is disabled. But if they are treated as blocking then checking these can be very expensive.

Reply:
Changed this to use `F.delay`. Thanks.